Classes/Core/Functional/Framework/Frontend/RequestBootstrap.php

```diff
@@ -160,6 +161,7 @@ public function executeAndOutput()
             chdir($_SERVER['DOCUMENT_ROOT']);
             SystemEnvironmentBuilder::run(0, SystemEnvironmentBuilder::REQUESTTYPE_FE);
             $container = Bootstrap::init($this->classLoader);
+            ClassLoadingInformation::registerClassLoadingInformation();
```

Comment:
Maybe we could add some notes why we would add this as Bootstrap::init() calls the same method as well (if in non-composer mode)

Reply:
It is aligned with the bootstrapping process of the FunctionalTestCase class [here](https://github.com/TYPO3/testing-framework/blob/master/Classes/Core/Testbase.php#L578) and [here](https://github.com/TYPO3/testing-framework/blob/master/Classes/Core/Testbase.php#L583). I would try to investigate why it was invented that way. Would you tell me, what you mean by non-composer setup: Is it the [Git + Composer Installation](https://get.typo3.org/version/10#accordion-git) (used for core development) or the [Wget / Curl / Download-Tar Installation](https://get.typo3.org/version/10#accordion-curl)?

Reply:
> non-composer setup = Git + Composer Installation

I tested the additional class loading with the Git + Composer installation as well and it showed that it was not required for that type of TYPO3 installation as the `Bootstrap::init()` one line above already loads those additional classes. Thus i added the condition `Environment::isComposerMode()` in order to handle that and to be in line with the `Bootstrap::init()` class load handling. 

To summarize: 
Functional tests in _composer mode_ and in _non-composer mode_ load the additional classes in `\TYPO3\TestingFramework\Core\Testbase::setUpBasicTypo3Bootstrap` via 
```
$this->dumpClassLoadingInformation();
``` 
– as some lines above in 
```
$container = Bootstrap::init($classLoader);
``` 
no composer autoload files are available yet and have to be dumped first.
In a frontend request, the regarding response intercepting class `RequestBootstrap` has to reload those additional classes which is done in `\TYPO3\TestingFramework\Core\Functional\Framework\Frontend\RequestBootstrap::executeAndOutput()` – for _non-composer mode_ in 
```
$container = Bootstrap::init($this->classLoader);
```
and for _composer mode_ in the code lines of this PR.

I also moved this loading of classes into a new method `Testbase::reloadClassLoadingInformation` to be aligned with method `Testbase::dumpClassLoadingInformation`.